schema_utils_test.py

```diff
@@ -474,6 +480,63 @@ def test_get_raises_invalid_validator_id(self):
         with self.assertRaises(Exception):
             schema_utils.get_validator('some invalid validator method name')
 
+    def test_is_valid_algebraic_expression_validator(self):
+        """Tests for the is_valid_math_expression static method with
+        algebraic type.
+        """
+        is_valid_math_expression = schema_utils.get_validator(
+            'is_valid_math_expression')
+
+        self.assertTrue(is_valid_math_expression('a+b*2'))
+        self.assertFalse(is_valid_math_expression('3+4/2'))
+
+    def test_is_valid_numeric_expression_validator(self):
+        """Tests for the is_valid_math_expression static method with
+        numeric type.
+        """
+        is_valid_math_expression = schema_utils.get_validator(
+            'is_valid_math_expression')
+
+        self.assertFalse(is_valid_math_expression('a+b*2', False))
+        self.assertTrue(is_valid_math_expression('3+4/2', False))
+
+    def test_is_valid_math_equation_validator(self):
+        """Tests for the is_valid_math_equation static method."""
+        is_valid_math_equation = schema_utils.get_validator(
+            'is_valid_math_equation')
+
+        self.assertTrue(is_valid_math_equation('a+b=c'))
+        self.assertTrue(is_valid_math_equation('x^2+y^2=z^2'))
+        self.assertTrue(is_valid_math_equation('y = m*x + b'))
+        self.assertTrue(is_valid_math_equation('alpha^a + beta^b = gamma^(-c)'))
+        self.assertTrue(is_valid_math_equation('a+b=0'))
+        self.assertTrue(is_valid_math_equation('0=a+b'))
+        self.assertTrue(is_valid_math_equation('(a/b)+c=(4^3)*a'))
+        self.assertTrue(is_valid_math_equation('2^alpha-(-3) = 3'))
+        self.assertTrue(is_valid_math_equation('(a+b)^2 = a^2 + b^2 + 2*a*b'))
+        self.assertTrue(is_valid_math_equation('x/a + y/b = 1'))
+        self.assertTrue(is_valid_math_equation('3 = -5 + pi^pi'))
+        self.assertTrue(is_valid_math_equation('pi = 3.1415'))
+        self.assertTrue(is_valid_math_equation('0.4 + 0.5 = alpha * 4'))
+        self.assertTrue(is_valid_math_equation('sqrt(a+b)=c - gamma/2.4'))
+        self.assertTrue(is_valid_math_equation('abs(35 - x) = 22.3'))
+
+        self.assertFalse(is_valid_math_equation('3 -= 2/a'))
+        self.assertFalse(is_valid_math_equation('x + y = '))
+        self.assertFalse(is_valid_math_equation('(a+b)^2 = a^2 + b^2 + 2ab'))
+        self.assertFalse(is_valid_math_equation('(a+b = 0)'))
+        self.assertFalse(is_valid_math_equation('a+b=0=a-b'))
+        self.assertFalse(is_valid_math_equation('alpha - beta/c'))
+        self.assertFalse(is_valid_math_equation('2^alpha-(-3*) = 3'))
+        self.assertFalse(is_valid_math_equation('a~b = 0'))
+        self.assertFalse(is_valid_math_equation('a+b<=0'))
```

Comment:
So, we are not supporting inequalities for now?

Reply:
That's right. Handling inequalities would be relatively complex and it's not that important as of now, which is why we'd decided to leave it for now.